concert/ext/ufo.py

```diff
@@ -282,6 +296,150 @@ def z_parameter(self, name):
+
+    """Data class holding arguments for tofu with units."""
+
+    UNITS = {
+        'source_position_y': q.px,
+        'detector_position_y': q.px,
+        'center_position_x': q.px,
+        'center_position_z': q.px,
+        'axis_angle_x': q.rad,
+        'energy': q.keV,
+        'propagation_distance': q.m,
+        'pixel_size': q.m,
+        'retrieval_padded_width': q.px,
+        'retrieval_padded_height': q.px,
+        'projection_margin': q.px,
+        'x_region': q.px,
+        'y_region': q.px,
+        'z': q.px,
+        'source_position_x': q.px,
+        'source_position_z': q.px,
+        'detector_position_x': q.px,
+        'detector_position_z': q.px,
+        'detector_angle_x': q.rad,
+        'detector_angle_y': q.rad,
+        'detector_angle_z': q.rad,
+        'axis_angle_y': q.rad,
+        'axis_angle_z': q.rad,
+        'volume_angle_x': q.rad,
+        'volume_angle_y': q.rad,
+        'volume_angle_z': q.rad,
+        'overall_angle': q.rad,
+        'y': q.px,
+        'height': q.px,
+        'width': q.px,
+    }
+
+    async def __ainit__(self, proxy, **kwargs):
+        # We keep a unitless version of the args to store the actual values
+        self._proxy = proxy
+        await super().__ainit__()
+
+        params = {}
+
+        for arg, settings in GeneralBackprojectArgs.parameters.items():
+
+            if arg in self.UNITS:
+                unit = self.UNITS[arg]
+                params[arg] = Quantity(
+                    unit,
+                    fget=self._make_getter(arg, unit=unit),
+                    fset=self._make_setter(arg, unit=unit),
+                    help=settings['help']
+                )
+            else:
+                params[arg] = Parameter(
+                    fget=self._make_getter(arg),
+                    fset=self._make_setter(arg),
+                    help=settings['help']
+                )
+
+        self.install_parameters(params)
+        await self._set_args(**kwargs)
+
+    def _make_getter(self, arg, unit=None):
+        async def getter(instance):
+            value = await self._proxy.get_reco_arg(arg)
+            if value is not None and unit is not None:
+                value *= unit
+
+            return value
+
+        return getter
+
+    def _make_setter(self, arg, unit=None):
+        async def setter(instance, value):
+            if unit is not None:
+                value = value.to(unit).magnitude
+                if isinstance(value, np.ndarray):
+                    value = value.tolist()
+
+            await self._proxy.set_reco_arg(arg, value)
+
+        return setter
+
+    async def _set_args(self, **kwargs):
+        for arg in kwargs:
+            if arg not in self._params:
+                raise AttributeError(
+                    f"QuantifiedGeneralBackprojectArgs do not have attribute `{arg}'"
+                )
+            await self._params[arg].set(kwargs[arg])
+
+    @property
```

Comment:
Could you make 'properties' concert Quantities/Parameters? Otherwise this will be confusing (since there is no async access, which is mandatory for the others to use).